tests/integration/golang/aim/metric/search_test.go

```diff
@@ -37,102 +40,6056 @@ func (s *SearchMetricsTestSuite) Test_Ok() {
+				metric1Run2,
+				metric2Run2,
+				metric3Run2,
+				metric1Run3,
+			},
+		},
+		{
+			name: "SearchMetricLastStepAndRunCreatedAtOperationLessOrEqualsAndGreater",
+			request: request.SearchMetricsRequest{
+				Query: `(((metric.name == "TestMetric1") or (metric.name == "TestMetric2") or (metric.name == "TestMetric3")) and (metric.last_step <= 3) and run.created_at > 111111111)`,
+			},
+
+			metrics: []*models.LatestMetric{
+				metric1Run1,
+				metric2Run1,
+				metric3Run1,
+				metric1Run3,
+			},
+		},
+		{
+			name: "SearchMetricLastStepAndRunCreatedAtOperationLessOrEqualsAndGreaterOrEquals",
+			request: request.SearchMetricsRequest{
+				Query: `(((metric.name == "TestMetric1") or (metric.name == "TestMetric2") or (metric.name == "TestMetric3")) and (metric.last_step <= 3) and run.created_at >= 111111111)`,
+			},
+
+			metrics: []*models.LatestMetric{
+				metric1Run1,
+				metric2Run1,
+				metric3Run1,
+				metric1Run2,
+				metric2Run2,
+				metric3Run2,
+				metric1Run3,
+			},
+		},
+		{
+			name: "SearchMetricLastStepAndRunCreatedAtOperationLessOrEqualsAndLess",
+			request: request.SearchMetricsRequest{
+				Query: `(((metric.name == "TestMetric1") or (metric.name == "TestMetric2") or (metric.name == "TestMetric3")) and (metric.last_step <= 3) and run.created_at < 222222222)`,
+			},
+
+			metrics: []*models.LatestMetric{
+				metric1Run1,
+				metric2Run1,
+				metric3Run1,
+				metric1Run2,
+				metric2Run2,
+				metric3Run2,
+			},
+		},
+		{
+			name: "SearchMetricLastStepAndRunCreatedAtOperationLessOrEqualsAndLessOrEquals",
+			request: request.SearchMetricsRequest{
+				Query: `(((metric.name == "TestMetric1") or (metric.name == "TestMetric2") or (metric.name == "TestMetric3")) and (metric.last_step <= 3) and run.created_at <= 222222222)`,
+			},
+
+			metrics: []*models.LatestMetric{
+				metric1Run1,
+				metric2Run1,
+				metric3Run1,
+				metric1Run2,
+				metric2Run2,
+				metric3Run2,
+				metric1Run3,
+			},
+		},
+		// Search complex query
+		{
+			name: "SearchMetricComplexQuery",
+			request: request.SearchMetricsRequest{
+				Query: `((metric.name == "TestMetric1") or (metric.name == "TestMetric2")) and metric.last_step >= 1 and (run.name.endswith("2") or re.match("TestRun1", run.name)) and (metric.last < 1.6) and run.duration > 0`,
+			},
+
+			metrics: []*models.LatestMetric{
+				metric1Run2,
+			},
+		},
+	}
+	for _, tt := range tests {
+		s.T().Run(tt.name, func(T *testing.T) {
+			var resp []byte
+			query, err := urlquery.Marshal(tt.request)
+			assert.Nil(s.T(), err)
+			resp, err = s.AIMClient.DoStreamRequest(
+				http.MethodGet,
+				fmt.Sprintf(`/runs/search/metric?%s`, query),
+				nil,
+			)
+			assert.Nil(s.T(), err)
+			decodedData, err := encoding.Decode(bytes.NewBuffer(resp))
+			assert.Nil(s.T(), err)
+
+			decodedMetrics := []*models.LatestMetric{}
+			for _, run := range runs {
+				metricCount := 0
+				for decodedData[fmt.Sprintf("%v.traces.%d.name", run.ID, metricCount)] != nil {
```

Comment:
i think this would look less cluttered if you a "prefix" of fmt.Sprintf(`%v.traces.%d`) and could then just use timestampsKey = prefix + "timestamps.blob", etc